src/Queue.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Equip\Queue;
+
+use Equip\Queue\Driver\DriverInterface;
+
+class Queue
+{
+    /**
+     * @var DriverInterface
+     */
+    private $driver;
+
+    /**
+     * @param DriverInterface $driver
+     */
+    public function __construct(DriverInterface $driver)
+    {
+        $this->driver = $driver;
+    }
+
+    /**
+     * Add message to queue
+     *
+     * @param string $queue
+     * @param string $name
+     * @param array $data
+     * @param array $meta
```

Comment:
It's expected that `$meta` may be modified by event listeners (e.g. to track job failure counts), where `$data` should remain the same in such circumstances.